projects/packages/assets/src/class-assets.php

```diff
@@ -64,18 +64,24 @@ public static function instance() {
 	 * Initalize the hooks as needed.
 	 */
 	private function init_hooks() {
+		global $wp_version;
 		/*
-		 * Load some scripts asynchronously.
+		 * Load some scripts asynchronously. Pre-WordPress 6.3 approach.
 		 */
-		add_filter( 'script_loader_tag', array( $this, 'script_add_async' ), 10, 2 );
+		if( version_compare( $wp_version, '6.3', '<' ) ) {
+			add_filter( 'script_loader_tag', array( $this, 'script_add_async' ), 10, 2 );
+		}
 	}
 
 	/**
 	 * A public method for adding the async script.
 	 *
+	 * @deprecated Since $$next-version$$, the `strategy` feature should be used instead.
+	 *
 	 * @param string $script_handle Script handle.
 	 */
 	public static function add_async_script( $script_handle ) {
+		_deprecated_function( __METHOD__, '$$next-version$$ ' );
 		$assets_instance                         = self::instance();
 		$assets_instance->defer_script_handles[] = $script_handle;
```

Comment:
You'll want to do something like this here so this keeps working in 6.3+.
```suggestion
		if ( version_compare( $wp_version, '6.3', '>=' ) ) {
			wp_script_add_data( $script_handle, 'strategy', 'defer' );
		} else {
			$assets_instance                         = self::instance();
			$assets_instance->defer_script_handles[] = $script_handle;
		}
```